test/robot_test.coffee

```diff
@@ -35,6 +35,7 @@ describe 'Robot', ->
 
     # Re-throw AssertionErrors for clearer test failures
     @robot.on 'error', (name, err, response) ->
+      return unless err?
```

Comment:
I changed it up to check the constructor as well, but I do always try to reduce nesting with early returns.